ibm/resource_ibm_is_instance_group_membership.go

```diff
@@ -0,0 +1,283 @@
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	instanceGroupID := d.Get("instance_group").(string)
+	instanceGroupMembershipID := d.Get("instance_group_membership").(string)
+
+	getInstanceGroupMembershipOptions := vpcv1.GetInstanceGroupMembershipOptions{
+		ID:              &instanceGroupMembershipID,
+		InstanceGroupID: &instanceGroupID,
+	}
+
+	instanceGroupMembership, response, err := sess.GetInstanceGroupMembership(&getInstanceGroupMembershipOptions)
+	if err != nil || instanceGroupMembership == nil {
+		if response != nil && response.StatusCode == 404 {
+			return nil
+		}
+		return fmt.Errorf("Error Getting InstanceGroup Membership: %s\n%s", err, response)
+	}
+
+	if v, ok := d.GetOk("action_delete"); ok {
+		actionDelete := v.(bool)
+		if actionDelete {
+			return resourceIBMISInstanceGroupMembershipDelete(d, meta)
+		}
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		name := v.(string)
+		if name != *instanceGroupMembership.Name {
+			d.SetId(fmt.Sprintf("%s/%s", instanceGroupID, instanceGroupMembershipID))
+
+			updateInstanceGroupMembershipOptions := vpcv1.UpdateInstanceGroupMembershipOptions{}
+			instanceGroupMembershipPatchModel := &vpcv1.InstanceGroupMembershipPatch{}
+			instanceGroupMembershipPatchModel.Name = &name
+
+			updateInstanceGroupMembershipOptions.ID = &instanceGroupMembershipID
+			updateInstanceGroupMembershipOptions.InstanceGroupID = &instanceGroupID
+			instanceGroupMembershipPatch, err := instanceGroupMembershipPatchModel.AsPatch()
+			if err != nil {
+				return fmt.Errorf("Error calling asPatch for InstanceGroupMembershipPatch: %s", err)
+			}
+			updateInstanceGroupMembershipOptions.InstanceGroupMembershipPatch = instanceGroupMembershipPatch
+			_, response, err := sess.UpdateInstanceGroupMembership(&updateInstanceGroupMembershipOptions)
+			if err != nil {
+				return fmt.Errorf("Error updating InstanceGroup Membership: %s\n%s", err, response)
+			}
+		}
+	}
+	return resourceIBMISInstanceGroupMembershipRead(d, meta)
+}
+
+func resourceIBMISInstanceGroupMembershipRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	instanceGroupID := parts[0]
+	instanceGroupMembershipID := parts[1]
+
+	getInstanceGroupMembershipOptions := vpcv1.GetInstanceGroupMembershipOptions{
+		ID:              &instanceGroupMembershipID,
+		InstanceGroupID: &instanceGroupID,
+	}
+	instanceGroupMembership, response, err := sess.GetInstanceGroupMembership(&getInstanceGroupMembershipOptions)
+	if err != nil || instanceGroupMembership == nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Getting InstanceGroup Membership: %s\n%s", err, response)
+	}
+	d.Set("delete_instance_on_membership_delete", *instanceGroupMembership.DeleteInstanceOnMembershipDelete)
```

Comment:
add d.setid in read method. If user passes only ins grp id and membership as fields to the resource, the tfstate should contain the details of that resource along with an id.